typespec-tests/src/main/java/com/type/model/inheritance/nesteddiscriminator/models/Shark.java

```diff
@@ -52,7 +52,6 @@ public String getSharktype() {
     public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
         jsonWriter.writeStartObject();
         jsonWriter.writeIntField("age", getAge());
-        jsonWriter.writeStringField("kind", getKind());
```

Comment:
Just notice serialziation and deserialization part, wonder why this get removed?